### PR TITLE
Add new ASHRAE single-zone VAV fan control for fan coil units

### PR DESCRIPTION
**Pull request by rraustad**
New feature for fan coil units per ASHRAE90.1 - SZVAV fan control.

### CODE REVIEW

File: doc/src/docs/InputOutputReference/01c-InputOutputReference.md [hunk 1]
```diff
@@ -7373,9 +7373,11 @@ EnergyPlus provides 5 capacity control methods for this unit:
...
 
-Note that the type of fan component associated with the fan coil unit depends on the type of capacity control method chosen. For *ConstantFanVariableFlow * a *Fan:OnOff* or *Fan:ConstantVolume* should be used. For *CyclingFan*, a *Fan:OnOff* should be used, for *VariableFanVariableFlow* or *VariableFanConstantFlow* a *Fan:VariableVolume*, and for *MultiSpeedFan* a *Fan:OnOff* should be chosen.
+Note that the type of fan component associated with the fan coil unit depends on the type of capacity control method chosen. For *ConstantFanVariableFlow * a *Fan:OnOff* or *Fan:ConstantVolume* should be used. For *CyclingFan*, a *Fan:OnOff* should be used, for *VariableFanVariableFlow* or *VariableFanConstantFlow* a *Fan:VariableVolume*, for *MultiSpeedFan* a *Fan:OnOff* should be used, and for *ASHRA90.1*, a *Fan:OnOff* or *Fan:VariableVolume* should be chosen.
```
Comment (mjwitte) [inline doc/src/docs/InputOutputReference/01c-InputOutputReference.md:7380]: Typo - ASHRA90.1

File: doc/src/docs/InputOutputReference/01c-InputOutputReference.md [hunk 2]
```diff
@@ -7323,9 +7323,11 @@ EnergyPlus provides 5 capacity control methods for this unit:
 
   5. multi-speed fan with cycling between speeds and  constant water flow.
 
+  6. fan speed control based on ASHRAE 90.1
```
Comment (rraustad) [inline doc/src/docs/InputOutputReference/01c-InputOutputReference.md:7326]: Missed this one (ASHRAE space 90.1)...wait, this one is OK

File: doc/src/docs/InputOutputReference/01c-InputOutputReference.md (hunk 1 above)
Comment (rraustad) [inline doc/src/docs/InputOutputReference/01c-InputOutputReference.md:7380]: and missed this one second time through

**Comment by rraustad** [comment]
How about if the comment for the commit started with NO CI TESTS, can that be a switch not to run the CI tests? I just wanted to push at work and pull at home to pick up where I left off.

File: doc/src/docs/InputOutputReference/01c-InputOutputReference.md (hunk 1 above)
Comment (mjwitte) [inline doc/src/docs/InputOutputReference/01c-InputOutputReference.md:7380]: And actually, now these all need to change to match what's in the IDD, ASHRAE90VariableFan

**Comment by rraustad** [comment]
Of course there's a warning that didn't show up on my end.

File: src/EnergyPlus/FanCoilUnits.cc [hunk 3]
```diff
@@ -1176,6 +1228,7 @@ namespace FanCoilUnits {
 		bool PrintFlag; // TRUE when sizing information is reported in the eio file
 		int SAFMethod( 0 ); // supply air flow rate sizing method (SupplyAirFlowRate, FlowPerFloorArea, FractionOfAutosizedCoolingAirflow, FractionOfAutosizedHeatingAirflow ...)
 		int CapSizingMethod( 0 ); // capacity sizing methods (HeatingDesignCapacity, CapacityPerFloorArea, FractionOfAutosizedCoolingCapacity, and FractionOfAutosizedHeatingCapacity )
+		bool SizingDesRunThisZone; // test for zone sizing
```
Comment (rraustad) [inline src/EnergyPlus/FanCoilUnits.cc:1237]: I should have also commented this out.

**Comment by Myoldmopar** [comment]
@rraustad I'm not sure what you're worried about with pushing to CI.  It doesn't hurt my feelings if CI is running on a few extra commits.  If you really want to avoid CI running your branch until you are ready, you can fork this repository and work in your own (github.com/rraustad/EnergyPlus) and then do a pull request over when you are ready.  This is how some of the team already works, and it is a fine way to work if you are worried about it.  Let me know if so and I can give some pointers on that workflow.

**Comment by Myoldmopar** [comment]
And it looks like that warning message is pretty simple to clean up.  Do you anticipate any additional work once that is cleaned up?

**Comment by rraustad** [comment]
I'd like to make the SA temp set point fields autosizable.
I'd also like to better protect the RegulaFalsi calls (i.e., a solution 
must be possible before you can call RegulaFalsi)

On 2/19/2016 9:16 AM, Edwin Lee wrote:

> And it looks like that warning message is pretty simple to clean up. 
> Do you anticipate any additional work once that is cleaned up?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/NREL/EnergyPlus/pull/5493#issuecomment-186232280.

## 

Richard Raustad
Florida Solar Energy Center
1679 Clearlake Road
Cocoa, FL 32922
Ph: (321)638-1454
http://www.fsec.ucf.edu/en/

Program Director
Electric Vehicle Transportation Center
http://evtc.fsec.ucf.edu/

**Comment by Myoldmopar** [comment]
OK @rraustad, I'll watch for an additional commit to drop in here.  If you decide you aren't going to be able to do it by today, then let me know so we can freeze without it.

**Comment by Myoldmopar** [comment]
@rraustad I changed the pull request title.  Is that reasonable?

**Comment by rraustad** [comment]
Perfect. Might add "for fan coil units" since that's as far as I got?

**Comment by Myoldmopar** [comment]
@rraustad I'll consider [this comment](https://github.com/NREL/EnergyPlus/pull/5493#issuecomment-186284031) as a list of to-do things in the scope of a followup fix for this feature.  That way the IO can be frozen for this branch today.  I'll be merging this shortly.  If you just keep working on your branch as if the Pull Request was never closed, it will be smooth.  We'll simply open a new pull request when you have new commits pushed up.

**Comment by rraustad** [comment]
I assume you are referring to this?

+// How to iterate on multiple independent variables, air flow rate, SA temperature, and water flow rate or elec heating coil PLR, to meet a specific zone load.

**Comment by Myoldmopar** [comment]
I meant the comment about making the SA temp field autosizable and protecting regula falsi calls.  But actually, on the autosizable front, we need to be very careful about **any** movement of the idd.  Comments, notes are fine, but field name changes and changes to functionality are almost certainly locked down for this release.

**Comment by rraustad** [comment]
For the Feb 19 commit, I was able to make SA Temp autosizable. I also revised the logic to protect RegulaFalsi calls but I could have missed something. No changes needed to the idd or code.

**Pull request merged**